__tests__/functional/tagged-format.test.ts

```diff
@@ -161,15 +161,15 @@ describe("tagged format", () => {
   // JS will actually fit big numbers into number but we use BigInt
   // any way so user can round trip longs.
   it.each`
-    input             | expected                  | expectedType | testCase
-    ${-(2 ** 31)}     | ${-(2 ** 31)}             | ${"number"}  | ${"-2**31"}
-    ${-(2 ** 31) - 1} | ${BigInt(-(2 ** 31) - 1)} | ${"bigint"}  | ${"-2**31 - 1"}
-    ${-(2 ** 63)}     | ${BigInt(-(2 ** 63))}     | ${"bigint"}  | ${"-2**63"}
-    ${2 ** 31 - 1}    | ${2 ** 31 - 1}            | ${"number"}  | ${"2**31 - 1"}
-    ${2 ** 31}        | ${BigInt(2 ** 31)}        | ${"bigint"}  | ${"2**31"}
-    ${2 ** 63 - 1}    | ${BigInt(2 ** 63 - 1)}    | ${"bigint"}  | ${"2**63 - 1"}
-    ${1.3 ** 63}      | ${1.3 ** 63}              | ${"number"}  | ${"1.3**63"}
-    ${1.3}            | ${1.3}                    | ${"number"}  | ${"1.3"}
+    input                             | expected                          | expectedType | testCase
+    ${-(2 ** 31)}                     | ${-(2 ** 31)}                     | ${"number"}  | ${"-2**31"}
+    ${-(2 ** 31) - 1}                 | ${BigInt(-(2 ** 31) - 1)}         | ${"bigint"}  | ${"-2**31 - 1"}
+    ${BigInt("-9223372036854775808")} | ${BigInt("-9223372036854775808")} | ${"bigint"}  | ${"-2**63"}
```

Comment:
i had this as a number intentionally as we seem to take a number and send it to Fauna as a long when its over 32 bits. It would then come out as a BigInt.

This looks likea  good additional case too.

But if my reading of the serializer / deserializer is correct we should keep that behavior as well.

Alghouth my tests was a bit crap since JS can't go over ~ 2 ^ 53 for numbers.

So maybe we add a case here for something like 2 ^ 40 and - (2 ^ 40) ?

Reply:
the number was too big for a fauna number, that's why I wrapped the input in BigInt.

Reply:
If the JS number is too big for an int, then it will become a double.

Reply:
That's not how the code reads:

https://github.com/fauna/fauna-js/blob/main/src/tagged-type.ts#L77-L79

and now that i'm looking again i'm not sure this code is safe. As we are calling 2 ^ 63 which exceeds: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER

Reply:
> the number was too big for a fauna number, that's why I wrapped the input in BigInt.

Please re-read my comment. I see that. I am suggesting an additional test.

Reply:
sorry I meant too big for long

Reply:
there's still some bugs in the source let's get those cleaned up and then we can cover the test cases

Reply:
okay, the problem was I saw that the int was too big to really be an int in JS so I was agressive about making it a BigInt. It should still work with the driver turning the JS number into a BigInt. No more argument, let's add that back.

Reply:
test appears to be failing for trying to compare BigInts with Jest `.toEqual` function. If I comment that out it passes.

Reply:
yeah but again i think we have a bug in the code.

The max safe int i is 2 ^ 53; and we are blowing past that in our source code.

So let's fix that up and also fix jest.

Reply:
comparing with .toString allows the tests to run, but of course passing in a js number too precise for a float truncates the value so we still need to adjust the test.

Reply:
Okay, we're there! `Math.isSafeInteger` to the rescue!! 🙏🏻

Reply:
Actually I am working on one more test case that throws the TypeError when the BigInt would lose precision.

Reply:
That's done, too.